docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/azure.md

```diff
@@ -110,6 +124,389 @@ Only hosts expected to be load balancer back ends need to be in this group.
+type: Opaque
+stringData:
+  cloud-config: |-
+    {
+      "cloud": "AzurePublicCloud",
+      "tenantId": "<tenant-id>",
+      "subscriptionId": "<subscription-id>",
+      "aadClientId": "<client-id>",
+      "aadClientSecret": "<tenant-id>",
+      "resourceGroup": "docker-machine-ks",
+      "location": "westus",
+      "subnetName": "docker-machine-ks",  
+      "securityGroupName": "rancher-managed-kqmtsjgJ",
+      "securityGroupResourceGroup": "docker-machine-ks",
+      "vnetName": "docker-machine-ks-vnet",
+      "vnetResourceGroup": "docker-machine-ks",
+      "primaryAvailabilitySetName": "docker-machine-ks",  
+      "routeTableResourceGroup": "docker-machine-ks",
+      "cloudProviderBackoff": false,
+      "useManagedIdentityExtension": false,
+      "useInstanceMetadata": true,
+      "loadBalancerSku": "standard",
+      "excludeMasterFromStandardLB": false,
+    }
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    kubernetes.io/cluster-service: "true"
+  name: system:azure-cloud-provider-secret-getter
+rules:
+  - apiGroups: [""]
+resources: ["secrets"]
+resourceNames: ["azure-cloud-config"]
+verbs:
+  - get
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    kubernetes.io/cluster-service: "true"
+  name: system:azure-cloud-provider-secret-getter
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:azure-cloud-provider-secret-getter
+  subjects:
+    - kind: ServiceAccount
+      name: azure-cloud-config
+      namespace: kube-system
+   ``` 
+
+## Using the out-of-tree Azure cloud provider
+
+<Tabs groupId="k8s-distro">
+<TabItem value="RKE2">
+
+1. Choose **External** from the Cloud Provider drop-down in the **Cluster Configuration** section. 
+
+2. - Prepare the Cloud Provider Configuration to set for the Cloud Provider Config in the next step. Note that Rancher will automatically create a new Network Security Group, Resource Group, Availability Set, Subnet, and Virtual Network. If you already have some or all of these created, you will need to specify them before creating the cluster. 
+   - You can click on "Show Advanced" to see more of these automatically generated names and update them if
+   necessary. Your Cloud Provider Configuration **must** match the fields in the Machine Pools section. If you have multiple pools, they must all use the same Resource Group, Availability Set, Subnet, Virtual Network, and Network Security Group.
+
+3. Under the **Cluster Configuration > Add-on Config** section, specify the `cloud-provider-azure` Helm chart in **Additional Manifest**. 
+
+Note that the chart reads the Cloud Provider Config from secret in `kube-system` namespace. An example secret for the Cloud Provider Config is shown below, modify it as needed. Please refer to the full list of configuration options in the [upstream docs](https://cloud-provider-azure.sigs.k8s.io/install/configs/).
```

Comment:
```suggestion
Note that the chart reads the Cloud Provider Config from the secret in the `kube-system` namespace. An example secret for the Cloud Provider Config is shown below. Modify it as needed. Refer to the full list of configuration options in the [upstream Azure docs](https://cloud-provider-azure.sigs.k8s.io/install/configs/).
```

Reply:
The suggested change has: `in the [upstream Azure docs]`, but the linked docs are provided by Kubernetes, not Azure.